src/d/actor/d_a_door10.cpp

```diff
@@ -6,170 +6,835 @@
 #include "d/actor/d_a_door10.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "f_op/f_op_actor_mng.h"
+#include "d/d_com_inf_game.h"
+#include "d/res/res_doorbs.h"
 
 /* 00000078-000000C8       .text chkMakeKey__10daDoor10_cFv */
-void daDoor10_c::chkMakeKey() {
-    /* Nonmatching */
+s32 daDoor10_c::chkMakeKey() {
+    u8 cVar1 = getType();
+    s32 ret;
+    s32 v = 2;
+
+    if (cVar1 == 4 || cVar1 == 5) {
+        ret = 1;
+    } else {
+        ret = cVar1 == 1 ? 2 : 0;
+    }
+    return ret;
```

Comment:
I think Roeming's suggestion is probably how the original devs wrote this, not a ternary https://decomp.me/scratch/QstUs
```suggestion
    if (cVar1 == 4 || cVar1 == 5) {
        return 1;
    }
    if (cVar1 == 1) {
        return 2;
    }
    return 0;
```